cirq/ops/gate_operation.py

```diff
@@ -41,6 +41,7 @@ def __init__(self,
             gate: The gate to apply.
             qubits: The qubits to operate on.
         """
+        gate.validate_args(qubits)
```

Comment:
Because this is now happening here, we should remove the call that is made during the `Gate.__call__` and `Gate.on` methods.